sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BoundedSource.java

```diff
@@ -133,6 +133,95 @@ public Double getFractionConsumed() {
     }
 
     /**
+     * A constant to use as the return value for {@link #getParallelismConsumed()} or
+     * {@link #getParallelismRemaining()} when the exact value is unknown.
+     */
+    public static final long PARALLELISM_UNKNOWN = -1;
+
+    /**
+     * Returns the total amount of parallelism in the consumed (and processed) range of this
+     * reader's current {@link BoundedSource} (as would be returned by {@link #getCurrentSource}).
+     *
+     * <p>The "parallelism" consumed does not necessarily equal the number of <em>records</em>
+     * output and processed, but rather the number of distinct (ideally non-empty) <em>sources</em>
```

Comment:
This description seems vague, but consumed parallelism is well-defined and unambiguous - it's the number of split points returned so far - 1 (or alternatively, number of processed "blocks" - ranges of records between two split points). I think it's best to refer here to RangeTracker for definition of split points.